### ISSUE TITLE
SQLite: Cannot parse foreign key keywords (DEFERRABLE, IMMEDIATE, INITIALLY)

### ISSUE THREAD

**Issue by mtlynch** [sqlfluff/sqlfluff#3716]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff doesn't seem to recognize the SQLite keywords related to [deferred enforcement of foreign keys](https://www.sqlite.org/foreignkeys.html#fk_deferred):

```sql
NOT DEFERRABLE INITIALLY DEFERRED
NOT DEFERRABLE INITIALLY IMMEDIATE
NOT DEFERRABLE
DEFERRABLE INITIALLY IMMEDIATE
DEFERRABLE
```

### Expected Behaviour

```bash
$ echo 'CREATE TABLE messages(msg_id TEXT, parent_id TEXT, FOREIGN KEY(parent) REFERENCES users(id) DEFERRABLE INITITALLY DEFERRED);' | sqlfluff lint --dialect sqlite -
All Finished 📜 🎉!
```

### Observed Behaviour

```bash
$ echo 'CREATE TABLE messages(msg_id TEXT, parent_id TEXT, FOREIGN KEY(parent) REFERENCES users(id) DEFERRABLE INITITALLY DEFERRED);' | sqlfluff lint --dialect sqlite -
== [stdin] FAIL                                                                                                                                                                                      
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE messages(msg_id TEXT, paren...'
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
```

### How to reproduce

```bash
$ echo 'CREATE TABLE messages(msg_id TEXT, parent_id TEXT, FOREIGN KEY(parent) REFERENCES users(id) DEFERRABLE INITITALLY DEFERRED);' | sqlfluff lint --dialect sqlite -
```

### Dialect

SQLite

### Version

```bash
$ sqlfluff --version
sqlfluff, version 1.2.1
$ python --version
Python 3.9.13
```

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
